packages/pubsub/src/subscription.js

```diff
@@ -149,154 +143,22 @@ var PUBSUB_API_TIMEOUT = 90000;
-     *     request.
-     * @param {?object} callback.metadata - Metadata of the subscription from
-     *     the API.
-     * @param {object} callback.apiResponse - Raw API response.
-     *
-     * @example
-     * subscription.getMetadata(function(err, metadata, apiResponse) {});
-     *
-     * //-
-     * // If the callback is omitted, we'll return a Promise.
-     * //-
-     * subscription.getMetadata().then(function(data) {
-     *   var metadata = data[0];
-     *   var apiResponse = data[1];
-     * });
-     */
-    getMetadata: {
-      protoOpts: {
-        service: 'Subscriber',
-        method: 'getSubscription'
-      },
-      reqOpts: {
-        subscription: this.name
-      }
-    }
-  };
+function Subscription(pubsub, name, options) {
+  options = options || {};
 
-  var config = {
-    parent: pubsub,
-    id: this.name,
-    methods: methods
-  };
+  this.pubsub = pubsub;
+  this.request = pubsub.request.bind(pubsub);
+  this.projectId = pubsub.projectId;
 
-  if (options.topic) {
-    // Only a subscription with knowledge of its topic can be created.
-    config.createMethod = pubsub.subscribe.bind(pubsub, options.topic);
-    delete options.topic;
-
-    /**
-     * Create a subscription.
-     *
-     * **This is only available if you accessed this object through
-     * {module:pubsub/topic#subscription}.**
-     *
-     * @param {object} config - See {module:pubsub#subscribe}.
-     *
-     * @example
-     * subscription.create(function(err, subscription, apiResponse) {
-     *   if (!err) {
-     *     // The subscription was created successfully.
-     *   }
-     * });
-     *
-     * //-
-     * // If the callback is omitted, we'll return a Promise.
-     * //-
-     * subscription.create().then(function(data) {
-     *   var subscription = data[0];
-     *   var apiResponse = data[1];
-     * });
-     */
-    config.methods.create = true;
-  }
+  this.name = Subscription.formatName_(pubsub.projectId, name);
+  this.connection = null;
+  this.encoding = options.encoding || 'utf-8';
+  this.ackDeadline = options.ackDeadline || 600;
 
-  commonGrpc.ServiceObject.call(this, config);
   events.EventEmitter.call(this);
 
-  this.autoAck = is.boolean(options.autoAck) ? options.autoAck : false;
-  this.closed = true;
-  this.encoding = options.encoding || 'utf-8';
-  this.inProgressAckIds = {};
-  this.interval = is.number(options.interval) ? options.interval : 10;
-  this.maxInProgress =
-    is.number(options.maxInProgress) ? options.maxInProgress : Infinity;
-  this.messageListeners = 0;
-  this.paused = false;
-
-  if (is.number(options.timeout)) {
-    this.timeout = options.timeout;
-  } else {
-    // The default timeout used in google-cloud-node is 60s, but a pull request
-    // times out around 90 seconds. Allow an extra couple of seconds to give the
-    // API a chance to respond on its own before terminating the connection.
-    this.timeout = PUBSUB_API_TIMEOUT + 2000;
+  if (options.topic) {
```

Comment:
Gotcha, that's something we do internally to let us know that we know about this subscriptions topic. Not something we tell the user to do. Generally if the user knows the topic we encourage them to do something like this

```js
var topic = pubsub.topic('my-topic');
var sub = topic.subscription('my-sub');
```